runtime/astar/src/lib.rs

```diff
@@ -1057,10 +1057,45 @@ pub type Executive = frame_executive::Executive<
     Migrations,
 >;
 
+pub use frame_support::traits::{OnRuntimeUpgrade, StorageVersion};
+pub struct HybridInflationModelMigration;
+impl OnRuntimeUpgrade for HybridInflationModelMigration {
+    fn on_runtime_upgrade() -> Weight {
+        let mut reward_config = pallet_block_rewards_hybrid::RewardDistributionConfig {
+            // 4.78%
+            treasury_percent: Perbill::from_rational(4_783_623u32, 100_000_000u32),
+            // 23.04%
+            base_staker_percent: Perbill::from_rational(23_041_451u32, 100_000_000u32),
+            // 17.27%
+            dapps_percent: Perbill::from_rational(17_272_878u32, 100_000_000u32),
+            // 3.06%
+            collators_percent: Perbill::from_rational(3_061_518u32, 100_000_000u32),
+            // 51.84%
+            adjustable_percent: Perbill::from_rational(51_840_530u32, 100_000_000u32),
+            // 60.00%
+            ideal_dapps_staking_tvl: Perbill::from_percent(60),
+        };
+
+        // This HAS to be tested prior to update - we need to ensure that config is consistent
+        #[cfg(feature = "try-runtime")]
+        assert!(reward_config.is_consistent());
+
+        // This should never execute but we need to have code in place that ensures config is consistent
+        if !reward_config.is_consistent() {
+            reward_config = Default::default();
+        }
```

Comment:
I wonder why the default value is used, instead of the current value in storage?

Reply:
I updated an enum field, so keeping the old enum will not work. If I use the current values there is a chance it is not sufficient (The same chance as the one I try to update - even though the vcalues has been unit tested).
Using Default is the most safe non-failing approach

Reply:
I see. It makes sense.